### ISSUE TITLE
SVG elements disappear

### ISSUE THREAD

**Issue by andraaspar** [MithrilJS/mithril.js#1820]
```JS
var data = {
  nextId: 0,
  ids: []
}

var App = {
  view: function (v) {
    return [
      m(`button`, {
        'onclick': () => {
          data.ids.push(data.nextId++)
        }
      },
        `Add item – Click me 4 times`
      ),
      m(`br`),
      m(`svg`,
        data.ids.slice(-2).map(id => (
          m(`g`, {
            'key': id,
          },
            m(`circle`, {
              'cx': 5 + 10 * id,
              'cy': 5,
              'r': 5,
            })
          )
        ))
      )
    ]
  }
}

m.mount(document.body, App)
```

Fiddle [here](https://jsfiddle.net/andraaspar/zy12oa4p/).

With the above code, you should see two circles appear and then crawl across the screen as you press the button multiple times.

Instead, you get two circles appear and then disappear one after the other. The weird thing is, the DOM inspector shows the two circles as part of the DOM – yet they are invisible.

Reproduced in both Chrome and Firefox.

**Comment by tivac** [comment]
Removing the `key` on the `<g>` fixes it for me, https://jsbin.com/zaqovi/1/edit?js,output

Haven't dug in any further though.

**Comment by andraaspar** [comment]
@tivac That's a workaround which I cannot use. I use inline SVG much like HTML, with components that are tied to the actual DOM element.

My workaround for now is to create multiple inline SVGs rather than try to model everything inside SVG – which is inefficient.

These are both just workarounds. [It should work:](https://mithril.js.org/hyperscript.html#svg-and-mathml)

> Mithril fully supports SVG.

This is a bug.

**Comment by andraaspar** [comment]
Maybe Mithril is creating the elements in the wrong document? (HTML document rather than SVG document?)

**Comment by isiahmeadows** [comment]
@andraaspar You *should* be able to check that via inspecting the output. And I wouldn't be surprised if that's the case - it's an easy thing to miss.

**Comment by edorrington** [comment]
It seems to have to do with the keys changing as you add items. In the fiddle, in your map over the data, if you set the key to be the index rather than the id (so that the keys remain the same on every iteration) it works fine. So, in other words, if instead of this:
`      m('svg',
      	data.ids.slice(-2).map(id => (
        	m('g', {
          	'key': id,
          },
`
you use this:
`      m('svg',
      	data.ids.slice(-2).map((id,index) => (
        	m('g', {
          	'key': index,
          },
`
then it works correctly.

**Comment by andraaspar** [comment]
@edorrington Reuse works, creating new elements is broken. This needs fixing.

**Comment by pygy** [comment]
@andraaspar no worries, we're on it

> Maybe Mithril is creating the elements in the wrong document? (HTML document rather than SVG document?)

Bingo!

**Comment by pygy** [comment]
@andraaspar fixed with #1864

**Comment by andraaspar** [comment]
@pygy Awesome job! Thank you!
